### PR TITLE
digital_ocean_floating_ip: check if fip is already assigned

### LINKED ISSUE #63175: digital_ocean_floating_ip is not idempotent
##### SUMMARY
`digital_ocean_floating_ip` is not idempotent, this is important.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
digital_ocean_floating_ip

##### ANSIBLE VERSION

```paste below
ansible 2.8.3
  config file = /Users/XXX/XXX/ansible.cfg
  configured module search path = ['/Users/XXX/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/XXX/XXX/.env/lib/python3.7/site-packages/ansible
  executable location = /Users/XXX/XXX/.env/bin/ansible
  python version = 3.7.4 (default, Jul  9 2019, 18:13:23) [Clang 10.0.1 (clang-1001.0.46.4)]
```

##### STEPS TO REPRODUCE
1. Run the tasks below. Droplet gets created with a Floating IP assigned.
2. Run the tasks once again, the playbook fails on the second step.

```yaml
- name: "Creating a droplet..."
  digital_ocean_droplet:
    state: present
    name: "{{ name }}"
    oauth_token: "{{ token }}"
    size: "{{ size }}"
    region: "{{ egion }}"
    image: "{{ image }}"
    wait_timeout: 300
    unique_name: true
    ipv6: true
    ssh_keys: "{{ ssh_key }}"
  register: digital_ocean_droplet

- name: "Create a Floating IP"
  digital_ocean_floating_ip:
    state: present
    oauth_token: "{{ token }}"
    droplet_id: "{{ digital_ocean_droplet.data.droplet.id }}"
  register: digital_ocean_floating_ip
```

##### EXPECTED RESULTS
Ansible detects that a Floating IP is already assigned to the droplet and returns `success` with the data included.

##### ACTUAL RESULTS

```paste below
TASK [cloud-digitalocean : Create a Floating IP] ***********************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error creating floating ip [422: Droplet is already assigned to another floating IP.]", "region": null}
```

### PR DESCRIPTION
**Pull request by jackivanov**
##### SUMMARY
We need to ensure that there is no floating ip assigned to the droplet already before trying to create a new one. Fixes #63175. Moved from #63179

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
digital_ocean_floating_ip

### CODE REVIEW

**Comment by jackivanov** [comment]
Sorry, I've messed up the git history, I'll rebase it

**Comment by jackivanov** [comment]
cc @pmarques

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py [hunk 1]
```diff
@@ -269,20 +269,39 @@ def associate_floating_ips(module, rest):
...
+
     if module.params['droplet_id'] is not None:
         payload["droplet_id"] = module.params['droplet_id']
 
-    response = rest.post("floating_ips", data=payload)
-    status_code = response.status_code
-    json_data = response.json
-    if status_code == 202:
-        module.exit_json(changed=True, data=json_data)
+        page = 1
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py]: Can you please use ? -https://github.com/ansible/ansible/blob/f21f3219b2a925a87e0c8aa54eb8ceca2cb55dc5/lib/ansible/module_utils/digital_ocean.py#L114

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py [hunk 2]
```diff
@@ -269,20 +269,39 @@ def associate_floating_ips(module, rest):
...
+        while page is not None:
+            response = rest.get('floating_ips?page={0}&per_page=20'.format(page))
+            json_data = response.json
+            if response.status_code == 200:
+                for floating_ip in json_data['floating_ips']:
+                    if floating_ip['droplet'] and floating_ip['droplet']['id'] == int(module.params['droplet_id']):
```
Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py]: Should we define the **droplet_id** param with _type_ **int** instead?

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py (hunk 1 above)
Comment (jackivanov) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py]: @Akasurde done

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py (hunk 2 above)
Comment (jackivanov) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip.py]: @pmarques done

**Comment by jackivanov** [comment]
@Akasurde can you take a look?

**Comment by Akasurde** [comment]
Hi @jackivanov, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.digitalocean<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**